src/cursor/types.d.ts

```diff
@@ -0,0 +1,3 @@
+export interface CursorState {
```

Comment:
Let's just put this in `cursor.ts` or `core_cursor.ts`. I'd prefer to only have a top-level `types.d.ts` if needed